mxnet-bot/LabelBotPredict/elastic_beanstalk_server/predict_labels.py

```diff
@@ -0,0 +1,106 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from sklearn.preprocessing import LabelEncoder
+from SentenceParser import SentenceParser
+from DataFetche import DataFetcher
+import numpy as np
+import pickle
+import re
+import logging
+
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+
+def tokenize(row):
+    row = re.sub('[^a-zA-Z0-9]', ' ', row).lower()
+    words = set(row.split())
+    return words
+
+def rule_based(row):
+    # return a list of rule_based labels
+    l = []
+    # 'feature request' in the issue's title
+    if "feature request" in row.lower():
+        l.append("Feature")
+    # 'c++' in the issue's title
+    if "c++" in row.lower():
+        l.append("C++")
+    tokens = tokenize(row)
+    ci_keywords = ["ci", "ccache", "jenkins"]
+    flaky_keywords = ["flaky"]
+    gluon_keywords = ["gluon"]
+    cuda_keywords = ["cuda", "cudnn"]
+    scala_keywords = ["scala"]
+    # one of ci keywords in the issue's title
+    for key in ci_keywords:
+        if key in tokens:
+            l.append("CI")
+            break
+    # one of flaky keywords in the issue's title
+    for key in flaky_keywords:
+        if key in tokens:
+            l.append("Flaky")
+    # one of gluon keywords in the issue's title
+    for key in gluon_keywords:
+        if key in tokens:
+            l.append("Gluon")
+    # one of scala keywords in the issue's title
+    for key in scala_keywords:
+        if key in tokens:
+            l.append("Scala")
+    # one of cudo keywords in the issue's title
+    for key in cuda_keywords:
+        if key in tokens:
+            l.append("CUDA")
+    return l
+
+def predict(issues):
```

Comment:
Can we split this portion into a class. It's not necessary to load model whenever you do the prediction. Instead we should try control load and update into a function so we can trigger once the auto-training is finished. This will also improve the prediction speed you have for now